### PR TITLE
Support ARM64 Linux MUSL target in release builds

### PR DESCRIPTION
**Pull request by jssblck**
# Overview

This PR refactors the build system to support the `aarch64-unknown-linux-musl` target in our release builds. It replaces cargo-dist with a custom GitHub Actions workflow that can handle all of our desired targets in parallel.

The PR also adds installer scripts for easy installation across all platforms and updates documentation accordingly.

Rendered README view: https://github.com/fossas/circe/tree/build/linux-arm64?tab=readme-ov-file#installation

## Acceptance criteria

- When a release tag is pushed, the new workflow will build all of our existing targets plus `aarch64-unknown-linux-musl`
- Developers can test the build process in PRs without creating actual GitHub Releases
- CI tests will run for our new target to ensure compatibility before releases
- Users can install circe with a simple one-liner on macOS, Linux, and Windows
- Installation scripts handle version compatibility and edge cases gracefully

## Testing plan

1. Create this PR and verify the new workflow runs successfully
2. Verify that all build artifacts are created correctly in the PR workflow run
3. Test the installation scripts on different platforms to ensure they work correctly
4. Once merged, push a tag to release a new version and verify all targets are built correctly

## Risks

- The new workflow removes some functionality from cargo-dist (like updater installation)
- Cross-compilation for ARM targets might have unexpected issues
- Installation scripts may need refinement after real-world usage

## References

- Part of https://fossa.atlassian.net/browse/ANE-2184
- Required for https://github.com/fossas/fossa-cli/pull/1514

## Checklist

- [x] I added tests for this PR's change (check-dynamic workflow now tests all targets)
- [x] I added cross-compilation support for ARM64 Linux platforms
- [x] I created installer scripts for all major platforms
- [x] I updated the README with new installation instructions

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+    fail "Neither sha256sum nor shasum found, cannot verify download"
+  fi
+
+  if [[ "$expected_checksum" != "$actual_checksum" ]]; then
+    fail "Checksum verification failed! Expected: $expected_checksum, got: $actual_checksum"
+  fi
+
+  # Extract archive
+  if [[ "$archive_name" == *.tar.gz ]]; then
+    tar -xzf "$archive_name"
```
Comment (csasarak) [inline install.sh:231]: I learned recently that most tars (not busybox apparently) will automatically detect the compression type if you just do it like `tar -xf`.

**Review by csasarak — APPROVED** [review]
